### PR TITLE
First working demo app

### PR DESCRIPTION
**Pull request by alxdrlitreev**

### CODE REVIEW

**Comment by andreev-io** [comment]
Let me know when this is ready for review.

**Comment by alxdrlitreev** [comment]
> Let me know when this is ready for review.

you may start

**Comment by andreev-io** [comment]
Reviewing 👍

File: Airtap/Airtap.entitlements [hunk 1]
```diff
@@ -2,9 +2,15 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
-    <key>com.apple.security.app-sandbox</key>
-    <true/>
-    <key>com.apple.security.files.user-selected.read-only</key>
-    <true/>
+	<key>com.apple.security.app-sandbox</key>
```
Comment (andreev-io) [inline Airtap/Airtap.entitlements:5]: Just so I know, what's the sandbox for?

File: Airtap/AppDelegate.swift [hunk 2]
```diff
@@ -8,30 +8,96 @@
 
 import Cocoa
 import SwiftUI
+import Combine
+import KeychainSwift
 
 @main
 class AppDelegate: NSObject, NSApplicationDelegate {
```
Comment (andreev-io) [inline Airtap/AppDelegate.swift:16]: nit: should there be an empty line here?

File: Airtap/Config.swift [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+//  Airtap
+//
+//  Created by Aleksandr Litreev on 28.02.2021.
+//  Copyright © 2021 Airtap OÜ. All rights reserved.
+//
+
+import Foundation
+
+struct Config {
+    static let apiEndpoint: String = "https://api-staging.airtap.dev/"
```
Comment (andreev-io) [inline Airtap/Config.swift:12]: What's a good way for us to deal with endpoint changes between staging and production environments in the future?

File: Airtap/Network/WS/WSService.swift [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        )
+        
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+        let bodyData = try! encoder.encode(payload)
+        let body = String(data: bodyData, encoding: .utf8)!
+        
+        print("SOCKETS OUT: \(body)")
+        
+        socket.write(string: body)
```
Comment (andreev-io) [inline Airtap/Network/WS/WSService.swift:126]: What happens if the underlying connection gets closed between lines 124 and 126? Will the code panic?

File: Airtap/Network/WebRTC/WebRTCService.swift [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    func createAnswer(for id: Int, completion: @escaping (String?) -> Void)
+    func setAnswer(for id: Int, sdp: String, completion: @escaping () -> Void)
+    func setCandidate(for id: Int, sdp: String, sdpMLineIndex: Int32, sdpMid: String?)
+}
+
+class WebRTCService: NSObject, WebRTCServing {
+    private let authProvider: AuthProviding
+    
+    private let rtcPeerConnectionFactory = RTCPeerConnectionFactory()
+    private let rtcMediaConstraints: RTCMediaConstraints = RTCMediaConstraints(
```
Comment (andreev-io) [inline Airtap/Network/WebRTC/WebRTCService.swift:40]: Not relevant to this PR, but we should research if there're any other constraints / settings we need to add here.

**Review by andreev-io — APPROVED** [review]
Everything looking good to me. I'm hitting approve. Please respond to my comments that I left mostly out of curiosity.

**Comment by andreev-io** [comment]
Completely forgot: before merging, please squash all your WIP commits:
`git rebase -i HEAD~30`

then mark the commits you want squashed with `s`.

File: Airtap/Airtap.entitlements (hunk 1 above)
Comment (alxdrlitreev) [reply Airtap/Airtap.entitlements:5]: It doesn't let our app interact with other apps. It is it's default value.

File: Airtap/Config.swift (hunk 3 above)
Comment (alxdrlitreev) [reply Airtap/Config.swift:12]: Yep, actually, good point. Can you deploy production and provide me with a domain name, so I can make it in a neaty way?

Comment (andreev-io) [reply Airtap/Config.swift:12]: Deployed to api.airtap.dev (note that the main branch is behind staging at the moment, so some API methods are buggy). TURN endpoints don't change between staging and production.

**Pull request closed without merging**